### PR TITLE
Fix controlplane components order

### PR DESCRIPTION
**Pull request by knrt10**
Also, added test to check the order

### CODE REVIEW

File: pkg/platform/platform_test.go [hunk 1]
```diff
@@ -45,3 +46,19 @@ func TestAppendVersionTag(t *testing.T) {
 		t.Fatalf("should append version tag to existing map")
 	}
 }
+
+func TestCommonControlPlaneChartsOrder(t *testing.T) {
+	expectedOrder := []string{"pod-checkpointer", "kube-apiserver", "kubernetes", "calico", "lokomotive", "bootstrap-secrets"} //nolint:lll
+
+	commonControlPlaneCharts := platform.CommonControlPlaneCharts()
```
Comment (invidian) [inline pkg/platform/platform_test.go:53]: Hooking on platform.CommonControlPlaneCharts is not the ideal place to check that, as the order may still be mangled by appending to the slice or even randomized before applying, but I think it's better than nothing.

File: pkg/platform/platform_test.go [hunk 2]
```diff
@@ -45,3 +46,19 @@ func TestAppendVersionTag(t *testing.T) {
 		t.Fatalf("should append version tag to existing map")
 	}
 }
+
+func TestCommonControlPlaneChartsOrder(t *testing.T) {
+	expectedOrder := []string{"pod-checkpointer", "kube-apiserver", "kubernetes", "calico", "lokomotive", "bootstrap-secrets"} //nolint:lll
```
Comment (invidian) [inline pkg/platform/platform_test.go:51]: Topic for other issue probably, but most likely `bootstrap-secrets` should be applied even before `pod-checkpointer`, so nodes can join the cluster for `pod-checkpointer` to converge.

**Review by invidian — APPROVED** [review]
Make sense

File: pkg/platform/platform_test.go (hunk 2 above)
Comment (surajssd) [reply pkg/platform/platform_test.go:51]: If we check if the worker nodes are up before applying `bootstrap-secrets` it will still fail.

Comment (invidian) [reply pkg/platform/platform_test.go:51]: I know, but that's a separate issue :)

File: pkg/platform/platform_test.go (hunk 1 above)
Comment (surajssd) [reply pkg/platform/platform_test.go:53]: I understand that new charts might be appended in the list. Like it is don for kubelet chart. But I don't know if the order will change as you say so. I see it is all ordered from there on.

**Review by surajssd — APPROVED** [review]
LGTM

File: pkg/platform/platform_test.go (hunk 1 above)
Comment (invidian) [reply pkg/platform/platform_test.go:53]: The point is, we do not check right before the applying, so it still can be mangled _somewhere_.

**Pull request merged**